inst/include/plant/disturbance_regime.h

```diff
@@ -0,0 +1,19 @@
+// -*-c++-*-
+#ifndef PLANT_PLANT_DISTURBANCE_REGIME_H_
+#define PLANT_PLANT_DISTURBANCE_REGIME_H_
+
+#include <vector>
+
+namespace plant {
+
+class Disturbance_Regime {
+public:
+  virtual double density(double time) const {return 0.0;};
+  virtual double pr_survival(double time) const {return 0.0;};
```

Comment:
@dfalster - this fixes the previous compilation problem, but means that `no_disturbance.h` below is pretty trivial.

Options I can see are either:
1. treating 'no disturbance' as the base case (as you suggested)
1. returning `NA_REAL` as the default

I don't mind either really, other than conceptual clarity of a base class with 'no disturbance' as a special case which side-steps the weirdness of 'Weibull disturbance' inheriting from 'no disturbance'.